Tiltfile

```diff
@@ -17,6 +17,7 @@ settings = {
     "enable_providers": ["docker"],
     "kind_cluster_name": os.getenv("CAPI_KIND_CLUSTER_NAME", "capi-test"),
     "debug": {},
+    "build_engine": "docker",
```

Comment:
might for a follow-up iteration, but wondering if we can auto-detect which build engine to use and ask the user to choose one only in case they have more than one (assuming this is possible, never tested personally).

Reply:
Yes, we can detect the engine by running the `<engine> version` command.

`docker version` can be podman on fedora with the package [podman-docker](https://packages.fedoraproject.org/pkgs/podman/podman-docker/)

So I propose to default to `"docker"` and change the default to `"podman"` if:
- `docker version` contains "Podman Engine"
- `docker version` fails but `podman version` contains "Podman Engine"


```diff
diff --git a/Tiltfile b/Tiltfile
index 70b7aaa56..d4c4b3d0f 100644
--- a/Tiltfile
+++ b/Tiltfile
@@ -4,12 +4,17 @@ envsubst_cmd = "./hack/tools/bin/envsubst"
 clusterctl_cmd = "./bin/clusterctl"
 kubectl_cmd = "kubectl"
 kubernetes_version = "v1.25.3"
+default_build_engine = "docker"

 if str(local("command -v " + kubectl_cmd + " || true", quiet = True)) == "":
     fail("Required command '" + kubectl_cmd + "' not found in PATH")

 load("ext://uibutton", "cmd_button", "location", "text_input")

+# detect if docker image should be built using podman
+if "Podman Engine" in str(local("docker version || podman version", quiet = True)):
+    default_build_engine = "podman"
+
 # set defaults
 version_settings(True, ">=0.30.8")

@@ -17,7 +22,7 @@ settings = {
     "enable_providers": ["docker"],
     "kind_cluster_name": os.getenv("CAPI_KIND_CLUSTER_NAME", "capi-test"),
     "debug": {},
-    "build_engine": "docker",
+    "build_engine": default_build_engine,
 }

 # global settings
```

It's working on my setup (podman on Fedora 37 with the podman-docker package), let me know if you want me to amend the commit to include this change.

Reply:
that's ok for me, I will test on my mac with docker as soon as ready

Reply:
I successfully tested the last version on a ubuntu 20.04 machine with docker 20.10.22